src/include/miopen/tensor_reorder_util.hpp

```diff
@@ -0,0 +1,286 @@
+};
+
+struct WrapperBatchedTransposeSolution_3012 : TensorReorderSolution
+{
+    BatchedTransposeSolution m_BatchedTransposeSolution;
+    WrapperBatchedTransposeSolution_3012(const ExecutionContext& ctx_,
+                                         miopenDataType_t data_type_,
+                                         uint32_t dim_0_,
+                                         uint32_t dim_1_,
+                                         uint32_t dim_2_,
+                                         uint32_t dim_3_)
+        : m_BatchedTransposeSolution(ctx_, data_type_, 1, dim_0_ * dim_1_ * dim_2_, dim_3_)
+    {
+    }
+    solver::KernelInfo GetKernel() const override { return m_BatchedTransposeSolution.GetKernel(); }
+    std::vector<OpKernelArg> GetKernelArg() const override
+    {
+        return m_BatchedTransposeSolution.GetKernelArg();
+    }
+    std::string GetKernelName() const override
+    {
+        return m_BatchedTransposeSolution.GetKernelName();
+    }
+    bool IsSkippable() const override { return m_BatchedTransposeSolution.IsSkippable(); }
+    size_t GetSize() const override { return m_BatchedTransposeSolution.GetSize(); }
+};
+
+struct WrapperGeneralReorderSolution : TensorReorderSolution
+{
+    GeneralReorderSolution m_GeneralReorderSolution;
+    WrapperGeneralReorderSolution(const ExecutionContext& ctx_,
+                                  miopenDataType_t data_type_,
+                                  uint32_t dim_0_,
+                                  uint32_t dim_1_,
+                                  uint32_t dim_2_,
+                                  uint32_t dim_3_,
+                                  uint32_t order_0_,
+                                  uint32_t order_1_,
+                                  uint32_t order_2_,
+                                  uint32_t order_3_)
+        : m_GeneralReorderSolution(ctx_,
+                                   data_type_,
+                                   dim_0_,
+                                   dim_1_,
+                                   dim_2_,
+                                   dim_3_,
+                                   order_0_,
+                                   order_1_,
+                                   order_2_,
+                                   order_3_)
+    {
+    }
+    solver::KernelInfo GetKernel() const override { return m_GeneralReorderSolution.GetKernel(); }
+    std::vector<OpKernelArg> GetKernelArg() const override
+    {
+        return m_GeneralReorderSolution.GetKernelArg();
+    }
+    std::string GetKernelName() const override { return m_GeneralReorderSolution.GetKernelName(); }
+    bool IsSkippable() const override { return m_GeneralReorderSolution.IsSkippable(); }
+    size_t GetSize() const override { return m_GeneralReorderSolution.GetSize(); }
+};
+
+__inline__ std::unique_ptr<TensorReorderSolution>
+TensorReorderSolutionConstructor(const ExecutionContext& ctx_,
```

Comment:
All the kernels written in HIP should be disabled when `ctx.use_hip_kernels` is `false`. The above means that all Solvers that use the new reorder/transpose kernels must become non-applicable.

Hence, this function should be able to inform the caller that it is unable to make the object. Note that this is similar to Solver::IsApplicable(), i.e. normal situation, not an error. Please extend the design and use this feature in test. As a result, the test must fail when `MIOPEN_DEBUG_HIP_KERNELS` is set to 0 in the environment.

Reply:
Resolved in #1481